packages/launch/src/focused-launch/summary/index.tsx

```diff
@@ -216,12 +218,32 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( { stepIndex } ) =>
 
 const Summary: React.FunctionComponent = () => {
 	const { title, updateTitle, saveTitle } = useTitle();
+	const [ forceSiteNameRender, setForceSiteNameRender ] = useState( false );
+
 	const { sitePrimaryDomain, siteSubdomain, hasPaidDomain } = useSiteDomains();
 	const selectedDomain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
 	const { setDomain, unsetDomain } = useDispatch( LAUNCH_STORE );
-
 	const domainSearch = useDomainSearch();
 
+	// The RegExp check is more relaxed on purpose — chances are that if the title
+	// matches (even it not exactly) the default title, the user would like want
+	// to change it before launch.
+	const isDefaultSiteTitle = new RegExp( DEFAULT_SITE_NAME, 'i' ).test( title );
+
+	// Ensure that the Site Name step doesn't disappear as soon as the user
+	// edits its value. This is achieved by ignoring the `isDefaultSiteTitle`
+	// check after the Site Name steps renders for the first time.
+	const shouldRenderNameStep = forceSiteNameRender || isDefaultSiteTitle;
+	useEffect( () => {
+		if ( shouldRenderNameStep ) {
+			setForceSiteNameRender( true );
+		}
+	}, [ shouldRenderNameStep ] );
```

Comment:
I hope this is not necessary 🙈 
See https://github.com/Automattic/wp-calypso/issues/47243#issuecomment-724934907

Reply:
As per #47243, I've added `isSiteTitleStepVisible` to the `automattic/launch` store. The selector and the action to be dispatch are made available through the `useTitle()` hook.

I also updated the testing instructions to reflect the need, for this new setting, to be persisted on a _per site_ base — although, for now, I'm not able to achieve that

Reply:
> to be persisted on a per site base

This should work inside the editor (when testing using editing-toolkit and sandbox). 

We should create an issue (not for the first release) to mention that we’ll need to use `siteId` when setting/getting all the items on the Launch store for persistence to work correctly in Calypso. So the store would probably look similar to the `Site` data-store https://github.com/Automattic/wp-calypso/blob/master/packages/data-stores/src/site/reducer.ts#L74

Reply:
I opened #47328 to track this task